overops-functions/src/main/java/com/takipi/udf/microsoftteams/MicrosoftTeamsChannelFunction.java

```diff
@@ -0,0 +1,195 @@
+            return;
+
+        ApiClient apiClient = args.apiClient();
+        UrlClient.Response<EventResult> eventResultResponse = getEventResultResponse(args, apiClient);
+        System.out.println("Got exceptionClassName, exception type, exceptionLocationPath, stack_frames, deployment");
+        MicrosoftTeamsUtil.EventData eventData = getEventData(apiClient, args);
+        System.out.println("Got applications, environments, deployments and servers");
+        MicrosoftTeamsUtil.TimeSlot timeSlot = getTimeSlot(MINUTES_TIME_SPAN);
+
+        SummarizedView view = getSummarizedView(rawContextArgs, args, apiClient);
+        System.out.println("Got viewId");
+
+        Graph eventsGraph = ViewUtil.getEventsGraph(apiClient, args.serviceId, view.id,
+                1, ValidationUtil.VolumeType.all, timeSlot.from, timeSlot.to);
+        System.out.println("Got machine_name, application_name, deployment_name");
+        String exceptionLinkToOverOps = EventUtil.getEventRecentLink(apiClient, args.serviceId, args.eventId,
+                timeSlot.from, timeSlot.to, eventData.applications,
+                eventData.servers, eventData.deployments);
+        System.out.println("Got ARC link");
+        MicrosoftTeamsChannelRequest microsoftTeamsChannelRequest = MicrosoftTeamsChannelRequest.newBuilder()
+                .setUrl(input.url)
+                .setEventResult(eventResultResponse.data)
+                .setExceptionLink(exceptionLinkToOverOps)
+                .setServer(eventsGraph.machine_name)
+                .setApplication(eventsGraph.application_name)
+                .setEnvironmentName(getEnvironmentName(eventData.environments, args.serviceId))
+                .setDeployment(eventsGraph.deployment_name)
+                .setDoNotAlertLink(getDoNotAlertLink(args, eventResultResponse.data.type))
+                .build();
+        UrlClient.Response<String> post = SimpleUrlClient.newBuilder().build().post(microsoftTeamsChannelRequest);
+        System.out.println("Post Microsoft Teams Webhook Channel request");
+
+        if (post.isBadResponse())
+            throw new IllegalStateException("Can't send card to " + input.url);
+    }
+
+    private static String getDoNotAlertLink(ContextArgs args, String exceptionType) {
+        return args.appHost + "/index.html?key=" + args.serviceId +
+                "&nav=archivemailitem&exception_class=" + Base64.getEncoder().encodeToString(exceptionType.getBytes());
+    }
+
+    private static UrlClient.Response<EventResult> getEventResultResponse(ContextArgs args, ApiClient apiClient) {
+        EventRequest eventRequest = EventRequest.newBuilder()
+                .setEventId(args.eventId)
+                .setServiceId(args.serviceId)
+                .setIncludeStacktrace(true)
+                .build();
+
+        UrlClient.Response<EventResult> eventResultResponse = apiClient.get(eventRequest);
+
+        if (!eventResultResponse.isOK() || eventResultResponse.data == null)
```

Comment:
Please have `{}` surrounding if code blocks, even if it's just a single line, as well as surround each `if` case in it's own parentheses
In this case:

```
if ((!eventResultResponse.isOK()) || (eventResultResponse.data == null)) {
    throw new IllegalStateException("Event is not present or api issue. eventId = " + args.eventId
                    + " , serviceId = " + args.serviceId);
}
```

Reply:
done